boost-python links against release python in debug builds

**Describe the bug**
Applications that use `boost-python` fail to link in debug build because `python*.lib` is not found. `python*_d.lib` should be used instead.
If I put `BOOST_DEBUG_PYTHON` into my custom triplet, `boost-python` build fails with `LINK : fatal error LNK1104: cannot open file 'python27_d.lib'`

**Environment**
- OS: Windows
- Compiler: MSVC v143

**To Reproduce**
Steps to reproduce the behavior:
1. `vcpkg install boost-python[python2,python3]:x64-windows`
2. Check runtime dependencies of `debug/bin/boost_python*.dll`.

**Expected behavior**
`boost-python` links against `python*_d.dll`, not `python*.dll`

--- a/ports/boost-modular-build-helper/CMakeLists.txt
+++ b/ports/boost-modular-build-helper/CMakeLists.txt
@@ -210,6 +210,15 @@ if(WITH_PYTHON3)
     list(APPEND python_versions "${python3_version}")
 endif()
 
+if(NOT python_versions STREQUAL "")
+    list(JOIN python_versions "," python_versions)
+    list(APPEND B2_OPTIONS "python=${python_versions}")
+
+    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+        list(APPEND B2_OPTIONS "python-debugging=on")
+    endif()
+endif()
+
 if("${PORT}" STREQUAL "boost-mpi" OR "${PORT}" STREQUAL "boost-graph-parallel")
     string(APPEND USER_CONFIG_EXTRA_LINES
         "using mpi : : <library-path>\"${CURRENT_INSTALLED_DIR}/lib\"\n"
@@ -224,11 +233,6 @@ if("${PORT}" STREQUAL "boost-mpi" OR "${PORT}" STREQUAL "boost-graph-parallel")
             "   <find-shared-library>openmpi ;\n"
         )
     endif()
-endif()
-
-if(NOT python_versions STREQUAL "")
-    list(JOIN python_versions "," python_versions)
-    list(APPEND B2_OPTIONS "python=${python_versions}")
 endif()
 
 # Include port specific CMake fragment

--- a/ports/boost-modular-build-helper/boost-modular-build.cmake
+++ b/ports/boost-modular-build-helper/boost-modular-build.cmake
@@ -154,6 +154,7 @@ function(boost_modular_build)
         string(REPLACE "-vc143-" "-vc140-" NEW_FILENAME ${NEW_FILENAME}) # To merge VS2022 and VS2015 binaries
         string(REPLACE "-sgd-" "-gd-" NEW_FILENAME ${NEW_FILENAME}) # For Debug libs
         string(REPLACE "-sgyd-" "-gyd-" NEW_FILENAME ${NEW_FILENAME}) # For Debug libs
+        string(REPLACE "-gyd-" "-gd-" NEW_FILENAME ${NEW_FILENAME}) # For Debug libs with python debugging
         string(REPLACE "-x32-" "-" NEW_FILENAME ${NEW_FILENAME}) # To enable CMake 3.10 and earlier to locate the binaries
         string(REPLACE "-x64-" "-" NEW_FILENAME ${NEW_FILENAME}) # To enable CMake 3.10 and earlier to locate the binaries
         string(REPLACE "-a32-" "-" NEW_FILENAME ${NEW_FILENAME}) # To enable CMake 3.10 and earlier to locate the binaries

--- a/ports/boost-modular-build-helper/vcpkg.json
+++ b/ports/boost-modular-build-helper/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "boost-modular-build-helper",
   "version": "1.79.0",
-  "port-version": 5,
+  "port-version": 6,
   "description": "Internal vcpkg port used to build Boost libraries",
   "license": "MIT",
   "dependencies": [

--- a/versions/b-/boost-modular-build-helper.json
+++ b/versions/b-/boost-modular-build-helper.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9428ba48caa81512147423f26559c7a57b37e71b",
+      "version": "1.79.0",
+      "port-version": 6
+    },
+    {
       "git-tree": "93946fc40b519f0f9afe1b0d36985b2def25d4ab",
       "version": "1.79.0",
       "port-version": 5

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -838,7 +838,7 @@
     },
     "boost-modular-build-helper": {
       "baseline": "1.79.0",
-      "port-version": 5
+      "port-version": 6
     },
     "boost-move": {
       "baseline": "1.79.0",

